### PR TITLE
webapp: Fixes to SSL commandlets

### PR DESCRIPTION
**Pull request by panchagnula**
## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [ ] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

**Comment by maddieclayton** [comment]
@panchagnula Looks like some of the tests need to be rerecorded, can you take a look?

**Comment by MiYanni** [comment]
No response in a couple weeks. Closing for now. Feel free to reopen when you have time to look into build failures/adding test recordings.

**Comment by panchagnula** [comment]
Looks like the PR moved to master as the base branch.. syncing & rebasing.

File: src/Websites/Websites/Cmdlets/Certificates/NewAzureWebAppSSLBinding.cs [hunk 1]
```diff
@@ -126,13 +126,12 @@ protected override void ProcessRecord()
                         pfxBlob: certificateBytes,
                         password: CertificatePassword,
                         hostingEnvironmentProfile: (webapp.HostingEnvironmentProfile != null) ?
```
Comment (cormacpayne) [inline src/Websites/Websites/Cmdlets/Certificates/NewAzureWebAppSSLBinding.cs]: @panchagnula any reason that we can't change this line to just be `hostingEnvironmentProfile: webapp.HostingEnvironmentProfile` ?

File: src/Websites/Websites/Utilities/WebsitesClient.cs [hunk 2]
```diff
@@ -820,6 +819,9 @@ public Site UpdateHostNameSslState(string resourceGroupName, string webAppName,
                 Location = location
             };
 
+            var webapp = GetWebApp(resourceGroupName, webAppName, slotName);
+            webappWithNewSslBinding.Tags = webapp.Tags;
```
Comment (cormacpayne) [inline src/Websites/Websites/Utilities/WebsitesClient.cs]: @panchagnula we should include a null check for the returned `webapp`, or would this fail somewhere before if the web app doesn't exist?

File: src/Websites/Websites/Cmdlets/Certificates/NewAzureWebAppSSLBinding.cs (hunk 1 above)
Comment (panchagnula) [reply src/Websites/Websites/Cmdlets/Certificates/NewAzureWebAppSSLBinding.cs]: hostingProfileEnvironment is set only for Apps On an ASE, in other cases this is null

Comment (panchagnula) [reply src/Websites/Websites/Cmdlets/Certificates/NewAzureWebAppSSLBinding.cs]: To answer your question, yes we can

File: src/Websites/Websites/Utilities/WebsitesClient.cs (hunk 2 above)
Comment (panchagnula) [reply src/Websites/Websites/Utilities/WebsitesClient.cs]: since the command is run on a webapp context webpp by itself will not be null

**Pull request merged**